beater/beater.go

```diff
@@ -193,12 +195,25 @@ func initTracer(info beat.Info, config *Config, logger *logp.Logger) (*elasticap
 		return nil, nil, err
 	}
 	if config.SelfInstrumentation.isEnabled() {
+		if config.SelfInstrumentation.Hosts != nil {
+			t, err := transport.NewHTTPTransport(config.SelfInstrumentation.Hosts[0], config.SelfInstrumentation.SecretToken)
+			if err != nil {
+				return nil, nil, err
+			}
+			tracer.Transport = t
+			logger.Infof("self instrumentation directed to %s", config.SelfInstrumentation.Hosts[0])
+		}
+
 		if config.SelfInstrumentation.Environment != nil {
 			tracer.Service.Environment = *config.SelfInstrumentation.Environment
 		}
 		tracer.SetLogger(logp.NewLogger("tracing"))
 	}
 
+	if config.SelfInstrumentation.isEnabled() && config.SelfInstrumentation.Hosts != nil {
```

Comment:
Should this instead be `if !config.SelfInstrumentation.isEnabled() || config.SelfInstrumentation.Hosts != nil {`? i.e. only set the pipe transport if self-instrumentation is enabled and there are no hosts specified.